### PR TITLE
TST: add regression test for apply case from GH-31605

### LINKED ISSUE #31605: Unexpected TypeError with groupby
#### Code Sample, a copy-pastable example if possible

```python
# Your code here
import numpy as np
import pandas as pd

def fct(group):
    return group[1].values.flatten()

df = pd.DataFrame([('A', 1), ('A', 2), ('A', 3), ('B', 4), ('B', 5), ('C', np.nan),])
df.groupby(0).apply(fct)
```
#### Problem description

This code works with pandas 0.25.1 BUT NOT with pandas 1.0.0.
With pandas 1.0.0, the output is:
TypeError: copy() takes no keyword arguments

#### Expected Output
With pandas 0.25.1, the output is:
Series:
0
A    [1.0, 2.0, 3.0]
B         [4.0, 5.0]
C              [nan]

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.1.final.0
python-bits      : 64
OS               : Windows
OS-release       : 7
machine          : AMD64
processor        : Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : fr_FR.cp1252

pandas           : 1.0.0
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 45.1.0.post20200127
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.11.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.1
numexpr          : None
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.3.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by fjetter**
@jorisvandenbossche you wanted to put more tests in for this issue. I put the user reported code example into a test. Did you have anything else in mind? I didn't merge it with the above one since the above acted on an empty Dataframe.

- [x] closes #31605
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/groupby/test_apply.py [hunk 1]
```diff
@@ -851,3 +851,16 @@ def test_apply_function_returns_non_pandas_non_scalar(function, expected_values)
     result = df.groupby("groups").apply(function)
     expected = pd.Series(expected_values, index=pd.Index(["A", "B"], name="groups"))
     tm.assert_series_equal(result, expected)
+
+
+def test_type_error_complex_index_non_scalar_return():
```
Comment (jorisvandenbossche) [inline pandas/tests/groupby/test_apply.py]:
```suggestion
def test_apply_function_returns_numpy_array():
```

File: pandas/tests/groupby/test_apply.py [hunk 2]
```diff
@@ -851,3 +851,16 @@ def test_apply_function_returns_non_pandas_non_scalar(function, expected_values)
...
+    # GH 31605
+    def fct(group):
+        return group[1].values.flatten()
+
+    df = pd.DataFrame([("A", 1), ("A", 2), ("A", 3), ("B", 4), ("B", 5), ("C", np.nan)])
+    result = df.groupby(0).apply(fct)
```
Comment (jorisvandenbossche) [inline pandas/tests/groupby/test_apply.py]: I know it's the example of the bug report, but can you give the dataframe column names? (eg ['a', 'b'])

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks! I think just this test is fine, added two minor comments

File: pandas/tests/groupby/test_apply.py (hunk 2 above)
Comment (fjetter) [reply pandas/tests/groupby/test_apply.py]: done

**Comment by jreback** [comment]
thanks @fjetter

**Pull request merged**